cpp/src/utilities/stacktrace.cpp

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/detail/utilities/stacktrace.hpp>
+
+#include <cxxabi.h>
+#include <execinfo.h>
+#include <stdlib.h>
+
+#include <sstream>
+
+namespace cudf::detail {
+
+std::string get_stacktrace(capture_last_stackframe capture_last_frame)
+{
+#ifdef __GNUC__
+  constexpr int max_stack_depth = 64;
+  void* stack[max_stack_depth];
+
+  auto const depth   = backtrace(stack, max_stack_depth);
+  auto const modules = backtrace_symbols(stack, depth);
+
+  if (modules == nullptr) { return "No stacktrace could be captured!"; }
+
+  std::stringstream ss;
+  size_t func_name_size = 256;
```

Comment:
That variable will be modified later in `__cxa_demangle` (line 72 below).

Reply:
This is awkward because the value is not used after modification. According to https://gcc.gnu.org/onlinedocs/libstdc++/libstdc++-html-USERS-4.3/a01696.html, you can pass `nullptr` and it won't write anything. So make this `constexpr` and pass `nullptr` instead of `&func_name_size` below.